jax/ad_util.py

```diff
@@ -43,13 +44,19 @@ def add_impl(xs, ys):
 def add_abstract(xs, ys):
   return lattice_join(xs, ys)
 
+def float0s_like(a):
+  return core.Float0Array(a.shape)
+
 jaxval_zeros_likers: Dict[type, Array] = {}
+jaxval_zeros_likers[dtypes.dtype(core.Float0Array)] = float0s_like
+jaxval_zeros_likers[dtypes.dtype(int)] = float0s_like
 
 def zeros_like_aval(aval):
   return aval_zeros_likers[type(aval)](aval)
 
 aval_zeros_likers: Dict[type, Array] = {}
```

Comment:
See comment below for why I think this type annotation wasn't precise enough to be helpful.

```suggestion
aval_zeros_likers: Dict[Type[core.AbstractValue], Array] = {}
```

Reply:
Updated type annotation.